### PR TITLE
Upgrade to golangci-lint v1.59.1, fix errors

### PR DESCRIPTION
**Pull request by aknuds1**
Upgrade golangci-lint from v1.59.0 to v1.59.1 and fix linting errors. I hit these errors locally, not sure if they are because I'm on v1.59.1.

### CODE REVIEW

File: discovery/eureka/client.go [hunk 1]
```diff
@@ -97,6 +97,7 @@ func fetchApps(ctx context.Context, server string, client *http.Client) (*Applic
 		resp.Body.Close()
 	}()
 
+	//nolint:usestdlibvars
```
Comment (machine424) [inline discovery/eureka/client.go:100]: I think we can open an issue on usestdlibvars for this.
I'd be a good improvement.

**Review by machine424 — APPROVED** [review]
Thanks!

File: discovery/eureka/client.go (hunk 1 above)
Comment (aknuds1) [reply discovery/eureka/client.go:100]: Thanks, I agree it would be best if they fixed this case. I'll open an issue with them.

Comment (aknuds1) [reply discovery/eureka/client.go:100]: @machine424 I can reproduce the problem with `usestdlibvars` v1.26.0, but it's apparently fixed in v1.27.0 (the latest).

**Pull request merged**